File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums):
        l = 1

        for r in range(1,len(nums)):
            if nums[r] != nums[r-1]:
                nums[l] = nums[r]
                l += 1
        return l
        

obj = Solution()
input_nums = [1,1,2]
result = obj.removeDuplicates(input_nums)
print(result, input_nums)
        

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public static int removeDuplicates(int[] nums) {
//         HashSet<Integer> set = new HashSet<>();
//         for (int i = 0; i < nums.length; i++) {
//             set.add(nums[i]);
//         }

//         int k = set.size();
//         int j = 0;
//         for (int x: set) {
//             nums[j++] = x;
// //            System.out.println(x);
//         }
//         return k;
        int i = 0;
        for (int j = 1; j < nums.length; j++) {
            if (nums[i] != nums[j]){
                i++;
                nums[i] = nums[j];
                // System.out.println(nums[j]);
            }
        }
        return i+1;

    }
}